spec/glueby/fee_provider/tasks_spec.rb

```diff
@@ -0,0 +1,506 @@
+RSpec.describe Glueby::FeeProvider::Tasks, active_record: true do
+  let(:tasks) { described_class.new }
+  let(:wallet) { TestInternalWallet.new }
+  let(:unspents) { [] }
+  let(:balance) { 0 }
+  let(:config) { {} }
+
+  before do
+    Glueby::FeeProvider.configure(config)
+    allow(Glueby::Internal::Wallet).to receive(:load).and_return(wallet)
+    fee_provider = Glueby::FeeProvider.new
+    tasks.instance_variable_set(:@fee_provider, fee_provider)
+
+    allow(wallet).to receive(:list_unspent).and_return(unspents)
+    allow(wallet).to receive(:balance).and_return(balance)
+  end
+
+  describe 'manage_utxo_pool' do
+    subject { tasks.manage_utxo_pool }
+
+    context 'UTXO pool is empty' do
+      context 'The wallet has enough TPC amount' do
+        let(:unspents) do
+          [{
+            txid: '5c3d79041ff4974282b8ab72517d2ef15d8b6273cb80a01077145afb3d5e7cc5',
+            script_pubkey: '76a914234113b860822e68f9715d1957af28b8f5117ee288ac',
+            vout: 0,
+            amount: 100_000_000,
+            finalized: true
+          }]
+        end
+        let(:balance) { 100_000_000 }
+
+        it 'makes the pool full' do
+          expect(wallet).to receive(:broadcast) do |tx|
+            expect(tx.inputs.count).to eq(1)
+            expect(tx.inputs[0].out_point).to eq(Tapyrus::OutPoint.from_txid('5c3d79041ff4974282b8ab72517d2ef15d8b6273cb80a01077145afb3d5e7cc5', 0))
+
+            expect(tx.outputs.count).to eq(21) # default UTXO pool size + a change output
+            expect(tx.outputs[0...20].map(&:value).uniq).to eq([1_000])
+            # The change output's value should equal to <before amount> - <UTXO pool size> * <fixed fee> - <fee of the tx>
+            expect(tx.outputs[20].value).to eq(100_000_000 - 20 * 1_000 - 1_000)
+          end
+          expect(tasks).to receive(:status)
+          subject
+        end
+
+        context 'fixed_fee option is specified' do
+          let(:config) { { fixed_fee: 800} }
+
+          it 'creates 800 tapyrus value outputs' do
+            expect(wallet).to receive(:broadcast) do |tx|
+              expect(tx.outputs[0...20].map(&:value).uniq).to eq([800])
+              # The change output's value should equal to <before amount> - <UTXO pool size> * <fixed fee> - <fee of the tx>
+              expect(tx.outputs[20].value).to eq(100_000_000 - 20 * 800 - 800)
+            end
+            expect(tasks).to receive(:status)
+            subject
+          end
+        end
+
+        context 'utxo_pool_size option is specified' do
+          let(:config) { { utxo_pool_size: 10 } }
+
+          it 'creates 10 outputs' do
+            expect(wallet).to receive(:broadcast) do |tx|
```

Comment:
You should check `tx.outputs.count` as well.

Reply:
done